core/corehttp/gateway_handler.go

```diff
@@ -525,21 +525,16 @@ func (i *gatewayHandler) deleteHandler(w http.ResponseWriter, r *http.Request) {
 	ctx, cancel := context.WithCancel(i.node.Context())
 	defer cancel()
 
-	p, err := path.ParsePath(urlPath)
+	p, err := coreapi.ParsePath(urlPath)
 	if err != nil {
 		webError(w, "failed to parse path", err, http.StatusBadRequest)
 		return
 	}
-
-	c, components, err := path.SplitAbsPath(p)
-	if err != nil {
-		webError(w, "Could not split path", err, http.StatusInternalServerError)
-		return
-	}
+	components := p.Components()[2:]
 
 	tctx, cancel := context.WithTimeout(ctx, time.Minute)
 	defer cancel()
-	rootnd, err := i.node.Resolver.DAG.Get(tctx, c)
+	rootnd, err := i.node.Resolver.DAG.Get(tctx, p.RootCid())
```

Comment:
Could it be possible to use the result of the previous `i.api.ResolveNode(ctx, rootPath)` changed up here which result get discarded?